tensorflow/python/ops/lookup_ops.py

```diff
@@ -1810,14 +1810,17 @@ def remove(self, keys, name=None):
 
     return op
 
-  def lookup(self, keys, name=None):
+  def lookup(self, keys, dynamic_default_values=None, name=None):
     """Looks up `keys` in a table, outputs the corresponding values.
 
     The `default_value` is used for keys not present in the table.
 
     Args:
       keys: Keys to look up. Can be a tensor of any shape. Must match the
         table's key_dtype.
+      dynamic_default_values: The values to use if a key is missing in the
```

Comment:
I think we need to define the semantics quite clearly here. Can I pass in a single value here? If there are multiple values I think the shape of this needs to be the same as keys? 

Also, I think users might expect to send in a single dynamic value - we should probably support that case?